compiler/runtime/OMRAOTRelocationRuntime.hpp

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_AOT_RELOCATION_RUNTIME_INCL
+#define OMR_AOT_RELOCATION_RUNTIME_INCL
+
+#ifndef OMR_AOT_RELOCATION_RUNTIME_CONNECTOR
+#define OMR_AOT_RELOCATION_RUNTIME_CONNECTOR
+namespace OMR { class OMRAOTRelocationRuntime; }
+namespace OMR { typedef OMR::OMRAOTRelocationRuntime OMRAOTRelocationRuntimeConnector; }
+#endif
+
+#include "runtime/TRRelocationRuntime.hpp"
+#include <map>
+#include <string>
+
+namespace TR { class RelocationRuntime; }
+
+namespace OMR
+{
+/**
+ * @brief OMRAOTRelocationRuntime holds a
+ * lookup by name map that would allow relocation runtime to have named
+ * symbol addresses and would allow linking external functions into runtime.
+ * The keys being strings is an intermediate in-development solution is
+ * supposed to change into something more suitable for a method
+ * identification, Resolved Method being a candidate for replacement. For
+ * now, maintaining the string being sufficient for a method verification
+ * is left as the runtime constructor responsibility.
```

Comment:
I question the name of this class. What about it is a "Runtime" ? What does it have to do with "Relocation" ? For that matter, what does it really have to do with "AOT"? It's a map from method names to `void *`'s, and it doesn't do much more than the underlying `std::map` does.

Maybe whoever currently has an AOTRelocationRuntime object can just hold the underlying std::map?

Reply:
Hello Mark @mstoodle , thank you for your comment(and the other comments too). This class is an attempt to demonstrate  extensible subclass of OMRRelocationRuntime that we're introducing, similar to what SharedCacheRelocationRuntime is to RelocationRuntime in OpenJ9. As OMRRelocationRuntime  might be used  by OpenJ9, that does not necessarily need the map, I made it a subclass. The map manipulation is done in this class for AOT project in OMR, hence the name.  The class that uses the map  functionality is AOTLoadStoreDriver (former AOTAdapter) that you were not fond of. I can try putting it there and getting rid of this class, and we could deal with AOTLoatStoreDriver later while reviewing this PR. Alternatively, we move some code from AOTLoadStoreDriver to this class,  to make the load store driver less "bag of things" as compared to what it is now. Can you please advise what would be the preferred way?

Reply:
I guess my confusion came from `OMRAOTRelocationRuntime` not overriding any behaviour of `TR::RelocationRuntime` nor calling down (up?) into `TR::RelocationRuntime`. It's adding seemingly unrelated code on top of a `TR::RelocationRuntime`. Based on that, perhaps moving the `std::map` into `AOTLoadStoreDriver` and having it talk directly to a `TR::RelocationRuntime` (if necessary) is the best approach rather than artificially plumping up this class.

Reply:
Hi @mstoodle , I've ended up moving the map to RelocationRuntime, the parent class. This map currently contains addresses of globals, addresses of memory regions and addresses of functions, that are requested by RelocationRecords from `Relocation Runtime`. I will push the changes after I will finish redesigning and rewiring `TR::AOTStorageInterface`